service/history/failover/coordinator.go

```diff
@@ -331,6 +364,7 @@ func (c *coordinatorImpl) notifyRemoteCoordinator(
 		if err != nil {
 			c.scope.IncCounter(metrics.FailoverMarkerNotificationFailure)
 			c.logger.Error("Failed to notify failover markers", tag.Error(err))
+			return
 		}
 
 		for marker := range requestByMarker {
```

Comment:
> this for loop just for clearing `requestByMarker`? Looks like we can just return an error from this function, and let the caller clear the map by assigning a new empty map to it instead of using a for loop?

Yes. Remove the loop here

Reply:
> Another question unrelated to this change: in the `aggregateNotificationRequests` func (L376) below, is the purpose to group `notificationRequest` by marker? But it looks like the key type of the map is a pointer to the marker instead of the marker itself. So if the marker used by different shards are different objects (I guess that's the case?), those markers won't actually be aggregated?

Yes. This is a bug. Changed it